src/Microsoft.AspNetCore.Mvc.Core/Internal/LocalRedirectResultExecutor.cs

```diff
@@ -42,7 +44,24 @@ public void Execute(ActionContext context, LocalRedirectResult result)
 
             destinationUrl = urlHelper.Content(result.Url);
             _logger.LocalRedirectResultExecuting(destinationUrl);
-            context.HttpContext.Response.Redirect(destinationUrl, result.Permanent);
+            if (!result.PreserveMethod)
```

Comment:
I have no particular opinion either way. Sometimes I like having the "most common" condition come first, sometimes I like avoiding the `!`.